### PR TITLE
Change loading order to avoid `cmp not found` when opening new file.

### PR DESCRIPTION
**Pull request by CharlesChiuGit**
and some minor improvements for loading order.
fix part of the issue from #395

### CODE REVIEW

**Comment by hhcwyx** [comment]
@CharlesChiuGit 按照[b95507d]的指导,修改了相关文件,一切正常了!!!非常感谢!谢谢手把手的精心指导,谢谢!

**Review by ayamir — APPROVED** [review]

File: lua/modules/completion/plugins.lua [hunk 1]
```diff
@@ -29,9 +29,9 @@ completion["hrsh7th/nvim-cmp"] = {
 	config = conf.cmp,
 	event = "InsertEnter",
 	requires = {
-		{ "onsails/lspkind.nvim" },
-		{ "lukas-reineke/cmp-under-comparator" },
-		{ "saadparwaiz1/cmp_luasnip", after = "LuaSnip" },
+		{ "onsails/lspkind.nvim", module = "lspkind" },
+		{ "lukas-reineke/cmp-under-comparator", module = "cmp-under-comparator" },
+		{ "saadparwaiz1/cmp_luasnip", after = { "nvim-cmp", "LuaSnip" } },
```
Comment (Jint-lzxy) [inline lua/modules/completion/plugins.lua:34]: @CharlesChiuGit A small question here, why `cmp_luasnip` is set both as a dependency of `nvim-cmp` and as being loaded after `nvim-cmp`?

File: lua/modules/editor/plugins.lua [hunk 2]
```diff
@@ -13,6 +13,11 @@ editor["terrortylor/nvim-comment"] = {
 }
 editor["nvim-treesitter/nvim-treesitter"] = {
 	opt = true,
+	module = {
+		"nvim-treesitter",
+		"nvim-treesitter.parsers",
+		"nvim-treesitter.ts_utils",
+	},
```
Comment (Jint-lzxy) [inline lua/modules/editor/plugins.lua:20]: Another problem: Directly opening a file by passing in command line parameters after this commit will cause cursor to be unable to move. After removing these lines, the problem disappears.

File: lua/modules/completion/plugins.lua (hunk 1 above)
Comment (CharlesChiuGit) [reply lua/modules/completion/plugins.lua:34]: @Jint-lzxy You're correct, there's not need to add "nvim-cmp". I just copy it from my own nvimdot. Thanks for correcting me.

File: lua/modules/editor/plugins.lua (hunk 2 above)
Comment (CharlesChiuGit) [reply lua/modules/editor/plugins.lua:20]: > Another problem: Directly opening a file by passing in command line parameters after this commit will cause cursor to be unable to move. After removing these lines, the problem disappears.

Interesting, I don't have this issue in the branch, but thanks for letting me know.

**Pull request merged**